### PR TITLE
WT-12078 Fix compact to handle pages in WT_REF_DELETED state

### PR DESCRIPTION
**Pull request by Sean04**
`test_compact12.py` has been updated to reopen the connection to ensure pages are on disk before truncating.

### CODE REVIEW

File: test/suite/test_compact12.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        # Remove 1/4 of the keys at timestamp 3.
+        c = self.session.open_cursor(uri, None)
+        for i in range(self.table_numkv // 4):
+            self.session.begin_transaction()
+            c.set_key(i)
+            c.remove()
+            self.session.commit_transaction(f'commit_timestamp={self.timestamp_str(3)}')
+        c.close()
+
+        self.session.checkpoint()
```
Comment (etienneptl) [inline test/suite/test_compact12.py:110]: I don't think this is needed as we are closing the connection which should perform a final checkpoint.

**Review by etienneptl — APPROVED** [review]
LGTM % s_all in the PR testing.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**